iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -278,23 +419,33 @@ bool ProcessManager::removeProcess(const ProcessName_t& name) noexcept
     auto it = m_processList.begin();
     while (it != m_processList.end())
     {
-        if (it->getName() == name)
+        auto otherName = it->getName();
+        if (name == otherName)
         {
-            m_portManager.deletePortsOfProcess(name);
-
-            m_processIntrospection->removeProcess(it->getPid());
-
-            // delete application
-            it = m_processList.erase(it);
-
-            LogDebug() << "New Registration - removed existing application " << name;
+            if (removeProcess(it))
+            {
+                LogDebug() << "New Registration - removed existing application " << name;
+            }
             return true; // we can assume there are no other processes with this name
         }
         ++it;
     }
     return false;
 }
 
+bool ProcessManager::removeProcess(ProcessList_t::iterator& processIter) noexcept
+{
+    // don't take the lock, else it needs to be recursive
```

Comment:
what does this mean? Do you expect this method to be called only with an already locked mutex? Calling erase on an unlocked list could end up in undefined behavior, or?

Reply:
I think this is only called from a method which already holds the lock. This could maybe made more clear by the method name.

Reply:
Agreed. Should we just call it `removeProcessWithoutLock`?
As @budrus said it's dangerous if it's not called from a method that already holds the lock. How should we address this? Add a comment in the description or in the method's name?

Reply:
See below. Added lock guard reference.